gammapy/datasets/tests/test_map.py

```diff
@@ -707,6 +707,62 @@ def test_map_fit(sky_model, geom, geom_etrue):
     assert not dataset_1.evaluators["test-model"].contributes
 
 
+@requires_data()
+@requires_dependency("ray")
+def test_map_fit_ray(sky_model, geom, geom_etrue):
+    from gammapy.datasets import DatasetsActor
+
+    dataset_1 = get_map_dataset(geom, geom_etrue, name="test-1")
+    dataset_2 = get_map_dataset(geom, geom_etrue, name="test-2")
+    datasets = Datasets([dataset_1, dataset_2])
+
+    models = Models(datasets.models)
+    models.insert(0, sky_model)
+
+    models["test-1-bkg"].spectral_model.norm.value = 0.5
+    models["test-model"].spatial_model.sigma.frozen = True
+
+    datasets.models = models
+    dataset_2.counts = dataset_2.npred()
+    dataset_1.counts = dataset_1.npred()
+
+    models["test-1-bkg"].spectral_model.norm.value = 0.49
+    models["test-2-bkg"].spectral_model.norm.value = 0.99
+
+    actors = DatasetsActor(datasets)
```

Comment:
In term of API, it is not possible to hide this line in the Fit class? A use of a 'parallel boolean' will allow to replace the list of datasets by the actor.... and then will lighten the user interface by avoiding to use very technical classes.
Do you get my point?

Reply:
Yes I did that in the begining but I removed it because : 
-if the datasets remain available in the main process and the  actors are recreated on each fit.run() call that's not very efficient
-if the datasets are replaced by actors in place (as now) it is better if the user is responsible of it because the API change to get and set attributes on the remote actors. First I don't think the setter should be implemented because attributes on the remote dataset should be immutable except for `models` (and maybe `mask_fit`), for example for now `datasets[0].counts = 1` run but does not apply to the remote dataset.
Secondly you gain the possibility to call (in parallel) any dataset attributes from the datasets like `datasets.counts()`,  `datasets.npred()`, `datasets.plot_residuals()`..., but the same could be done with the current datasets for consistency.

Ideally the `MapDatasetActor` should be have a method `from_maker` to fill them direclty in their respective processes so data are not copied. I'm not sure yet if this should be done through the `DatasetsMaker` or if the `DatasetsActor` should have a `from_maker` method that behave differently.

Reply:
I improved the get and set mecanism.
Now the remote class is automatically updated, and trying to change anything except models and mask_fit raise an error.